YouVsVirus/Assets/Scripts/Infection/InfectionManager.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+using UnityEngine;
+using Components;
+
+namespace Infection
+{
+    /// <summary>
+    /// Manage different infection models.
+    /// Infection models have to be attached to humans.
+    /// </summary>
+    public class InfectionManager : MonoBehaviour
+    {
+        /// <summary>
+        /// Agent based time delayed infection model
+        /// </summary>
+        private TimeDelayInfection tdi;
```

Comment:
i´d prefer longer variable names then pdi/tdi.
timeDelayModel and probabilityBasedModel seem good choices.